libs/ddhub-client-gateway-frontend/ui/channels/src/lib/containers/modals/Create/Restrictions/effects/roleRestriction.effects.ts

```diff
@@ -37,17 +38,32 @@ export const useRolesRestrictionEffects = (currentRoles: string[]) => {
   };
 
   const addRole = (role: string) => {
-    if (roles.includes(role)) {
-      return;
+    if (!roles.includes(role)) {
+      const rolesToSet = [role, ...roles];
+      setRoles(rolesToSet);
     }
-    setRoles([...roles, role]);
+
     clearRolesInput();
   };
 
   const removeRole = (value: string) => {
     setRoles(roles.filter((role) => role !== value));
   };
 
+  const updateRole = (type: RestrictionType, oldValue: string, newValue: string) => {
+    const filteredRoles = roles.filter((role) => role !== oldValue);
+
+    if (!filteredRoles.includes(newValue)) {
+      if (type === RestrictionType.Role) {
```

Comment:
btw. is this if statement really needed? The method is already telling that you need to use Role, not DID. So you know you're using Role. This way you could also remove `type` property from function declaration.

Reply:
Yes, it is needed. I raised a concern related to this as well, and we've still decided to do it. 

Before this, we cannot edit `did` or `role` at all. Now we can edit not just the value, but the `type` as well. Both `did` and `role` will have to perform the update. Example: change `type` from `did` to `role`, then we must remove `oldValue` from `did` array and add `newValue` to `role` array.

The design treat the list is as if it is a single `restriction` list, while it is not. Hence that's why the code becomes complicated.

Reply:
Ok, thanks for clarify :)